pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1445,6 +1450,37 @@ self: super: {
   # binary-instances needs the latest version.
   time-compat = self.time-compat_1_9_4;
 
+  # - Deps are required during the build for testing and also during execution,
+  #   so add them to build input and also wrap the resulting binary so they're in
+  #   PATH.
+  # - This requires many overrides because of the bumped
+  #   optparse-applicative version impacting the test-suite. Once
+  #   optparse-applicative-0.16.0.0 (or newer) is the default then this (and
+  #   the overrides) can be removed.
+  update-nix-fetchgit = let deps = [ pkgs.git pkgs.nix pkgs.nix-prefetch-git ];
+  in generateOptparseApplicativeCompletion "update-nix-fetchgit" (overrideCabal
+    (super.update-nix-fetchgit.override rec {
+      optparse-generic = self.optparse-generic_1_4_4;
+      optparse-applicative = self.optparse-applicative_0_16_0_0;
+      hnix = self.hnix.override { inherit optparse-applicative; };
+      tasty = self.tasty.override { inherit optparse-applicative; };
+      tasty-golden = self.tasty-golden.override {
+        inherit tasty optparse-applicative;
+        tasty-hunit = self.tasty-hunit.override { inherit tasty; };
+      };
+    }) (drv: {
+      buildTools = drv.buildTools or [ ] ++ deps ++ [ pkgs.makeWrapper ];
```

Comment:
Is this correct?  Does `update-nix-fetchgit` really need `git` and `nix` as a buildtool?

Reply:
They're used for the tests. A fake git repo and nix store are created to test updating expressions based on the result of nix-prefetch-git

Reply:
> They're used for the tests.

In that case, I imagine they should probably go in `testToolDepends`?

Reply:
Ah of course, thanks. I've added this now, along with `addTestToolDepends` to `lib.nix`